packages/host/app/components/operator-mode/code-submode.gts

```diff
@@ -682,14 +688,47 @@ export default class CodeSubmode extends Component<Signature> {
         ></div>
       </:ready>
     </RealmInfoProvider>
-    <CardURLBar
-      @loadFileError={{this.loadFileError}}
-      @resetLoadFileError={{this.resetLoadFileError}}
-      @userHasDismissedError={{this.userHasDismissedURLError}}
-      @dismissURLError={{this.dismissURLError}}
-      @realmURL={{this.realmURL}}
-      class='card-url-bar'
-    />
+    <div class='code-mode-top-bar'>
+      <CardURLBar
+        @loadFileError={{this.loadFileError}}
+        @resetLoadFileError={{this.resetLoadFileError}}
+        @userHasDismissedError={{this.userHasDismissedURLError}}
+        @dismissURLError={{this.dismissURLError}}
+        @realmURL={{this.realmURL}}
+        class='card-url-bar'
+      />
+      <BoxelDropdown>
+        <:trigger as |bindings|>
+          <Button
+            {{bindings}}
+            @kind='primary'
+            @size='small'
+            class='new-file-button'
+            data-test-new-file-button
+          >
+            <IconPlus
+              @width='var(--boxel-icon-sm)'
+              @height='var(--boxel-icon-sm)'
+              stroke='var(--boxel-light)'
+              stroke-width='1px'
+              alt='Add'
+              class='new-file-button-icon'
+            />
+            New File
+          </Button>
+        </:trigger>
+        <:content as |dd|>
+          <Menu
+            @items={{array
+              (menuItem
+                'Card Instance' this.toggleNewInstanceModal disabled=true
+              )
+            }}
+            @closeMenu={{dd.close}}
+          />
+        </:content>
+      </BoxelDropdown>
```

Comment:
Since we have a goal of trimming down the size of code-submode.gts, let's make this it's own component in `packages/host/app/components/operator-mode/code-submode/new-file-button.gts`